unified-runtime/source/loader/layers/sanitizer/msan/msan_interceptor.hpp

```diff
@@ -225,20 +225,27 @@ struct USMLaunchInfo {
   ur_context_handle_t Context = nullptr;
   ur_device_handle_t Device = nullptr;
   const size_t *GlobalWorkSize = nullptr;
-  const size_t *GlobalWorkOffset = nullptr;
+  std::vector<size_t> GlobalWorkOffset;
   std::vector<size_t> LocalWorkSize;
   uint32_t WorkDim = 0;
 
   USMLaunchInfo(ur_context_handle_t Context, ur_device_handle_t Device,
                 const size_t *GlobalWorkSize, const size_t *LocalWorkSize,
                 const size_t *GlobalWorkOffset, uint32_t WorkDim)
       : Data(Context, Device), Context(Context), Device(Device),
-        GlobalWorkSize(GlobalWorkSize), GlobalWorkOffset(GlobalWorkOffset),
-        WorkDim(WorkDim) {
+        GlobalWorkSize(GlobalWorkSize), WorkDim(WorkDim) {
     if (LocalWorkSize) {
       this->LocalWorkSize =
           std::vector<size_t>(LocalWorkSize, LocalWorkSize + WorkDim);
     }
+    // UR doesn't allow GlobalWorkOffset is null, we need to construct a zero
+    // value array if user doesn't specify its value.
+    if (GlobalWorkOffset) {
```

Comment:
global work offset can be null, see:
https://github.com/intel/llvm/blob/d10b37e131340272a1487bd1355de82a2835a746/unified-runtime/include/ur_api.h#L7857-L7859

If something is segfaulting when null is passed to launch kernel, please file a bug or let me know.